src/components/ContactChip.tsx

```diff
@@ -14,58 +19,65 @@ import { getContactURL } from "@utils/helpers/contact";
 
 const ContactChip = ({
   contact,
+  allowEdit,
+  onDelete,
   className,
 }: {
   contact: Contact;
+  allowEdit?: boolean;
+  onDelete?: () => void;
   className?: string;
 }): JSX.Element => {
   const locale = useRouter().locale as LangCode;
 
   return (
-    <a
-      href={getContactURL(contact.type, contact.value)}
-      target="_blank"
-      rel="noreferrer"
-    >
-      <Card
-        type="horizontal"
-        appearance="tonal"
-        hasAction
-        className={className}
-      >
-        <CardHeader
-          icon={<ContactIcon icon={contact.type} width={24} />}
-          title={
+    <Card type="horizontal" appearance="tonal" hasAction className={className}>
+      <CardHeader
+        icon={<ContactIcon icon={contact.type} width={24} />}
+        title={
+          <a
+            href={getContactURL(contact.type, contact.value)}
+            target="_blank"
+            rel="noreferrer"
+          >
             <span id={`contact-${contact.id}`}>{contact.name[locale]}</span>
-          }
-          end={
-            contact.includes ? (
-              <div className="flex flex-row-reverse">
-                {contact.includes.teachers && (
-                  <MaterialIcon
-                    icon="school"
-                    className="text-stroke text-stroke-0.25 text-stroke-surface-1 -ml-2 text-secondary"
-                  />
-                )}
-                {contact.includes.parents && (
-                  <MaterialIcon
-                    icon="escalator_warning"
-                    className="text-stroke text-stroke-0.25 text-stroke-surface-1 -ml-2 text-primary"
-                  />
-                )}
-                {contact.includes.students && (
-                  <MaterialIcon
-                    icon="groups"
-                    className="text-stroke text-stroke-0.25 text-stroke-surface-1 -ml-2 text-primary"
-                  />
-                )}
-              </div>
-            ) : undefined
-          }
-          className="bg-surface-1 !p-2"
-        />
-      </Card>
-    </a>
+          </a>
+        }
+        end={
+          <div className="flex flex-row-reverse">
+            {allowEdit && (
+              <Button
+                type="text"
+                icon={<MaterialIcon icon="delete" />}
+                iconOnly
+                className=" ml-4"
+                isDangerous
+                onClick={onDelete}
+              />
+            )}
+            {contact.includes!.teachers && (
```

Comment:
`includes` can be undefined if the Contact is not for a group (i.e. a teacher’s email). I know that the Contact Chip has not been used for this purpose yet (even though now that I think of it it is pretty important), it is best not to remove support for it.